csFastFloat/Utils/Utils.cs

```diff
@@ -66,7 +73,52 @@ internal static bool is_integer(char c, out uint cMinus0)
       cMinus0 = cc;
       return res;
     }
-    
+
+
+
+#if HAS_INTRINSICS
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    unsafe internal static bool eval_parse_eight_digits_simd(char* start, char* end, out uint value)
+    {
+
+      value = 0;
+
+      Vector128<short> raw = Sse41.LoadDquVector128((short*)start);
+      Vector128<short> ascii0 = Vector128.Create((short)47);
+      Vector128<short> after_ascii9 = Vector128.Create((short)58);
+
+      var a = Sse41.CompareGreaterThan(raw, ascii0);
+      var b = Sse41.CompareLessThan(raw, after_ascii9);
+      var c = Sse41.Subtract(a, b);
```

Comment:
Right so with unsigned arithmetic, we can do (c - '0') <= '9'-'0'. With signed arithmetic, we must do something more complicated... (c - '0' + short.MinValue) <= '9'-'0' +  short.MinValue where short.MinValue == - 2^15 or something of the sort. So if c = '0'...'9', the left hand side will be between short.MinValue and '9'-'0' +  short.MinValue.

@CarlVerret I think you can save one instruction there.

Reply:
To use TestZ I have to add 1 to previous vector.  Means an additionnal operation

Reply:
> To use TestZ I have to add 1 to previous vector. Means an additionnal operation
 
You are right that you have to be careful to avoid needing yet another instruction and thus wiping out your gains... but you can apply TestZ on 9 < something (just reverse your parameters).

In any case, looking at the assembly output should resolve the matter... either you save an instruction or you do not and that's easy to see when looking at the assembly instructions (just count!).